patches/tModLoader/Terraria/ModLoader/IO/WorldIO.cs

```diff
@@ -662,7 +694,25 @@ private static TagCompound SaveModHeaders()
 
 		var saveData = new TagCompound();
 		foreach (var system in SystemLoader.Systems) {
-			system.SaveWorldHeader(saveData);
+			try {
+				system.SaveWorldHeader(saveData);
+			}
+			catch {
+				var message = NetworkText.FromKey("tModLoader.SaveWorldHeaderExceptionWarning", system.Name, system.Mod.Name);
+				Logging.tML.Error(message.ToString());
+
+				if (Main.gameMenu && Main.menuMode == 10) {
+					// Save and Quit. Since this isn't on main thread, attempting Interface.errorMessage.Show behaves poorly because the menu will automatically change while the user is reading the message. For now don't do anything.
+				}
+				else if (!Main.gameMenu) {
+					// In-game autosave
+					ChatHelper.BroadcastChatMessage(message, Color.OrangeRed); // Handles SP and Server cases.
+				}
+
```

Comment:
Can we commonise this code with `PlayerIO`? I think `ChatHelper.BroadcastChatMessage` doesn't actually broadcast if called from client so no need for two different `NewText` calls